### PR TITLE
Run CI with Python 3.8

### PR DESCRIPTION
**Pull request by tai271828**
This pull request is the first milestone to resolve Issue: #238 . The approach of this CI job is based on the new features of `devenv` https://github.com/solvcon/devenv/pull/88 and mixes the build solution with `conda` and `devenv` in the meantime.

### CODE REVIEW

File: .github/workflows/solvcon_install.yml [hunk 1]
```diff
@@ -79,8 +99,29 @@ jobs:
         which python3
         python3 --version
         python3 -c 'import numpy ; print("numpy.__version__:", numpy.__version__)'
+        which gmsh
+
+    - name: show python env
```
Comment (tai271828) [inline .github/workflows/solvcon_install.yml:104]: Simply printing environment information for us to trace the importing configuration when mixing `conda` and `devenv`.

File: contrib/conda.sh [hunk 2]
```diff
@@ -1,11 +1,10 @@
 #!/bin/bash
 conda install -y \
-  python=3.6 \
+  python=3.8 \
   cmake setuptools pip sphinx ipython jupyter \
   cython numpy hdf4 netcdf4 nose pytest paramiko boto graphviz
 lret=$?; if [[ $lret != 0 ]] ; then exit $lret; fi
-conda install -y -c https://conda.anaconda.org/yungyuc \
-  gmsh scotch
+conda install -y -c https://conda.anaconda.org/yungyuc scotch
```
Comment (tai271828) [inline contrib/conda.sh:7]: `gmsh` has been supported and introduced in the build script.

File: .github/workflows/solvcon_install.yml [hunk 3]
```diff
@@ -30,9 +30,19 @@ jobs:
         sudo apt-get -qqy install fakeroot debhelper locales \
                 libreadline7 libssl1.0.0 libffi6 \
                 liblapack3 liblapack-dev libhdf5-100 libhdf5-dev libnetcdf13 \
-                libnetcdf-dev libscotch-6.0 libscotch-dev cython3 python3 \
```
Comment (tai271828) [inline .github/workflows/solvcon_install.yml:33]: cython will be built by `devenv`.

File: setup.py [hunk 4]
```diff
@@ -112,6 +112,7 @@ def make_cython_extension(
     if extra_compile_args is None: extra_compile_args = []
     extra_compile_args = [
         '-Werror',
+        '-Wno-deprecated-declarations', # issue: 238
```
Comment (tai271828) [inline setup.py:115]: Inspired by the discussion of cython https://github.com/solvcon/solvcon/issues/238#issuecomment-932166404 . This workaround is applied for macOS.

**Review by yungyuc — APPROVED** [review]
Great!

How come I didn't approve it in the first place?  Sorry for missing this!

**Comment by tai271828** [comment]
I am glad that you were having a nice holiday : )  Thanks for merging!

**Pull request merged**